src/main/java/com/hp/octane/plugins/bamboo/listener/OctanePostChainAction.java

```diff
@@ -243,5 +253,25 @@ public void execute(Chain chain, ChainResultsSummary chainResultsSummary, ChainE
         com.atlassian.bamboo.v2.build.BuildContext buildContext = (BuildContext) chainExecution.getBuildIdentifier();
         ParametersHelper.addParametersToEvent(ciEvent, buildContext);
         OctaneSDK.getClients().forEach(client -> client.getEventsService().publishEvent(ciEvent));
+
+        String buildCiId =
+                PlanKeys.getPlanResultKey(chain.getPlanKey(), chainExecution.getBuildIdentifier().getBuildNumber())
+                        .getKey();
```

Comment:
And also here I think to good to add try catch for IllegalArguments execption like you wrote me